### ISSUE TITLE
dotnet test3 orphans vstest process when cancelled with ctrl+c

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#4504]
## Steps to reproduce
1. Create  core app
2. Execute "dotnet test3" against a project that doesn't have a test framework. Output will show the following and hang indefinitely
   
   ```
   Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
   Copyright (c) Microsoft Corporation.  All rights reserved.
   
   Starting test execution, please wait...
   ```
3. CTRL+C to exit
## Expected  behavior

Should kill all processes, including vs console and msbuild processes
## Actual behavior

Top-level process prints "Attempting to cancel the build..." and exits, but child process continues to execute.

The orphaned process is 

```
"dotnet.exe" exec C:\Users\namc\.dotnet\sdk\1.0.0-preview3-003913\MSBuild.dll /t:VSTest /verbosity:quiet /nologo C:\Users\namc\dev\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.Tests\Microsoft.DotNet.Watcher.Tools.Tests.csproj
```

MSBuild will periodically spit out the following to stdout

```
C:\Users\namc\.dotnet\sdk\1.0.0-preview3-003913\Microsoft.TestPlatform.targets(37,5): warning MSB4220: Waiting for the currently executing task "Microsoft.TestPlatform.Build.Tasks.VSTestTask" to cancel. [C:\Users\namc\dev\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.Tests\Microsoft.DotNet.Watcher.Tools.Tests.csproj]
```
## Environment data

Microsoft.NET.Sdk/1.0.0-alpha-20161026-3
cli version 1.0.0-preview3-003913
win10-x64

**Comment by livarcocc** [comment]
@balachir @Faizan2304 Can you guys take a look?

**Comment by natemcmaster** [comment]
Talked with @Faizan2304. It seems there are at least two things that lead to this. (1) using stale packages and versions of CLI, and (2) trying to run a netcoreapp1.1 test project (the version of vstest.console I'm using is only netcoreapp1.0).

**Comment by piotrpMSFT** [comment]
I talked to @Faizan2304 last night and understood that this issue has been fixed in the VSTest repo and that a new drop will be coming to CLI this week.

**Comment by jgoshi** [comment]
Does this still repro with the latest stuff? When I try running dotnet test on a net core app it doesn't hang and I get this (which seems correct?):
E:\_dotnetcli_samples\test> dotnet test
Build started, please wait...
Build completed.

Test run for E:\_dotnetcli_samples\test\bin\Debug\netcoreapp1.0\test.dll(.NETCoreApp,Version=v1.0)
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Error: Could not find testhost.dll for source 'E:\_dotnetcli_samples\test\bin\Debug\netcoreapp1.0\test.dll'. Make sure test project has a nuget reference of package "microsoft.testplatform.testhost".

**Comment by natemcmaster** [comment]
@jgoshi you're probably hitting https://github.com/Microsoft/vstest/issues/287, which is not the same as this issue. You can workaround by adding `<OutputType>exe</OutputType>` to your project.

**Comment by jgoshi** [comment]
Can you still repro this? According to the comments from 12/5 it looks like this might be fixed by now? I couldn't repro it (cancelling seemed to work for me). Unless I'm doing the repro steps wrong.

**Comment by natemcmaster** [comment]
It reproduced in 1.0.0-preview4-004233 but appears to be fixed in 1.0.0-preview5-004468.

Here are my test steps:

1. dotnet new -t xunittest
2. Add `while(true);` to the test method in Tests.cs
3. dotnet test
4. CTRL+C

Result:
When orphaned, the console begins to prints out this line every ~10 seconds, indefinitely

> C:\Program Files\dotnet\sdk\1.0.0-preview4-004233\Microsoft.TestPlatform.targets(32,5): warning MSB4220: Waiting for the currently executing task "Microsoft.TestPlatform.Build.Tasks.VSTestTask" to cancel.

**Comment by jgoshi** [comment]
Yeah I tried your repro steps as well and it looks like this is fixed. Closing the issue. Please re-open if the issue resurfaces.
